flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/plan/batch/table/BatchPythonScalarFunctionSplitRuleTest.scala

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.batch.table
+
+import org.apache.flink.api.common.typeinfo.{BasicTypeInfo, TypeInformation}
+import org.apache.flink.api.scala._
+import org.apache.flink.table.api.scala._
+import org.apache.flink.table.functions.{FunctionLanguage, ScalarFunction}
+import org.apache.flink.table.planner.utils.TableTestBase
+import org.junit.Test
+
+class BatchPythonScalarFunctionSplitRuleTest extends TableTestBase {
```

Comment:
Rule test should be moved to directory rules/logical. There is no need to test both Stream and Batch cases.

Reply:
Maybe we can rename it to PythonCalcTest and add this PythonScalarFunctionSplitRuleTest in the rules/logical. The PythonCalcTest is used to test `BatchExecPythonCalcRule` and the whole optimization phase while PythonScalarFunctionSplitRuleTest is used to test the `PythonScalarFunctionSplitRule`.

In other words, we will have three tests, i.e., two PythonCalcTest(one for batch and one for stream), one PythonScalarFunctionSplitRuleTest